internal/utils/write_file.go

```diff
@@ -22,3 +24,33 @@ func WriteFile(path string, contents []byte, executable bool) error {
 
 	return nil
 }
+
+/* Writes a temp file and flushes the buffer */
+func CreateAndWriteTempFile(fileName string, data []byte) (string, error) {
+	f, err := os.CreateTemp("", fileName)
```

Comment:
We can but we would need to create the subdirectory. I had tried to do this with os.Mkdir but even with setting the permissions to 0666 I was getting issues with reading from the file in unit tests and when running worker. Maybe I missed something? If we instead to MkTempDir however that ends up generating a randomized directory each time.

Reply:
Where you using `t.TempDir` during tests?

Reply:
I was getting access issues when trying to read the temp file during tests as well as when running worker. I think I was able to get this working now though. In this case we can still create the files in the temp directory but we don't need to use CreateTemp and generate a random file name. Since each worker writes to their own local storage space we shouldn't have issues with file names being the same. We also wouldn't need to save the file paths anymore and instead just save the sha256 hashes and add on the directory when we write to zip. Any thoughts?